python/graphstorm/model/gnn_encoder_base.py

```diff
@@ -131,14 +131,33 @@ def dist_inference_one_layer(layer_id, g, dataloader, target_ntypes, layer, get_
     -------
     dict of Tensors : the inferenced tensors.
     """
+    len_dataloader = max_num_batch = len(list(dataloader))
+    barrier()
+    tensor = th.tensor([len_dataloader], device=device)
+    if is_distributed():
+        th.distributed.all_reduce(tensor, op=th.distributed.ReduceOp.MAX)
+        max_num_batch = tensor[0]
+
+    dataloader_iter = iter(dataloader)
     y = {}
-    for iter_l, (input_nodes, output_nodes, blocks) in enumerate(dataloader):
+
+    # To use WholeGraph for feature featching, dataloaders from different
+    # trainers must iterate through the same number of iterations as WholeGraph
+    # does not support imbalanced batch numbers across processes/trainers
+    # TODO (IN): Fix dataloader to have same number of minibatches
+    for iter_l in range(max_num_batch):
+        if iter_l < len_dataloader:
+            input_nodes, output_nodes, blocks = next(dataloader_iter)
```

Comment:
If input_nodes do not contain all the ntypes in g, we need to add those missing ntypes as ntype:th.empty(0,).

Reply:
Is this necessary?

Reply:
Are you sure?

Reply:
Yes

Reply:
Does our dataloader guarantee that it will sample all the node types in a minibatch?
If in a minibatch, not all the node types have nodes been sampled, will the dataloader add an empty tensor for that node type?

Lets either add an assertion or do a 
```
    for ntype in g.ntypes:
``` 
to check input_nodes[ntype]

Reply:
Same foe edge_gnn.py

Reply:
Confirmed. All minibatches have all the node types. Added assert as an additional check.